.github/workflows/deploy-frontend.yml

```diff
@@ -16,14 +25,21 @@ jobs:
         run: |
           echo "Running on branch ${{ github.ref }}"
           if [ "${{ github.ref }}" = "refs/heads/master" ]; then
-            echo "##[set-output name=branch;]master"
+            echo "branch=master" >> $GITHUB_OUTPUT
           else
-            echo "##[set-output name=branch;]staging"
+            echo "branch=staging" >> $GITHUB_OUTPUT
           fi
         id: extract_branch
+      - name: Generate revert command
+        id: generate-revert-command
+        env:
+          BRANCH: ${{ steps.extract_branch.outputs.branch }}
+        run: |
+          aws amplify list-jobs --app-id $APP_ID --branch-name $BRANCH --max-items 1 | jq '.jobSummaries[0]' | echo
+          JOB_ID=$(aws amplify list-jobs --app-id $APP_ID --branch-name $BRANCH --max-items 1 | jq '.jobSummaries[0].jobId')
+          echo "revert_cmd=aws amplify start-job --app-id $APP_ID --branch-name $BRANCH --job-id $JOB_ID --job-type RETRY" >> $GITHUB_OUTPUT
```

Comment:
I'm new to GitHub Actions workflows but my understanding is that for each component (backend, frontend, etc.) we're generating a revert command by querying the current code revision deployed for each component. I wonder whether doing this on a component-by-component basis could cause inconsistencies, where we accidentally revert different components to different code revisions. Is there any way we can query for the last revision successfully deployed, and generate the revert commands based on that, to ensure all components will be reverted to the same revision.

NB There's a reasonable chance I misunderstood the code, and this isn't an issue.

Reply:
yup I share this concern too, but right now there's no way to figure out `last successful deployment` that's why I'm resorting to this

Reply:
But I dont think it's a big concern cos we will be notified of failed deployments and will revert the code before the next deployment attempt